### ISSUE TITLE
OCI-SDK Version Upgradation Issue v2.41.0 to v3.1.0

### ISSUE THREAD

**Issue by ganeshshindegit** [oracle/oci-java-sdk#468]
Hi Team, For Helidon MP (2.5.4) microservice, I am not able to connect to the OCI vault after upgrading the OCI SDK version from 2.41.0 to 3.1.0, getting the exception Caused by: com.oracle.bmc.model.BmcException: Error returned by Unknown operation operation in Unknown service.(-1, null, false) Unknown error

Code Snippet for fetching the secretes from the OCI vault:
```
final InstancePrincipalsAuthenticationDetailsProvider provider = InstancePrincipalsAuthenticationDetailsProvider.builder().build();
            final Secrets secrets = SecretsClient.builder().build(provider);
            final SecretBundleContentDetails content = secrets.getSecretBundleByName(GetSecretBundleByNameRequest.builder()
                    .secretName(secreteName)
                    .vaultId(OCI_VAULT_ID)
                    .build())
                    .getSecretBundle()
                    .getSecretBundleContent();
```
I have attached the complete error stack trace file. I have also raised this topic on Oracle's internal slack channel.
[OCI-SDK-Error Stack Trace.pdf](https://github.com/oracle/oci-java-sdk/files/10373308/OCI-SDK-Error.Stack.Trace.pdf)

**Comment by mricken** [comment]
There was a regression in Instance Principal auth in version 3.1.0 that was fixed in 3.2.0.

Please upgrade to version 3.2.1 or later.

**Comment by mricken** [comment]
Confirmed to work in `3.2.0` and higher.
